src/gui/qgsmediawidget.h

```diff
@@ -0,0 +1,104 @@
+/***************************************************************************
+  qgsmediawidget.h
+
+ ---------------------
+ begin                : 2023.01.24
+ copyright            : (C) 2023 by Mathieu Pellerin
+ email                : mathieu at opengis dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSMEDIAWIDGET_H
+#define QGSMEDIAWIDGET_H
+
+#include <QWidget>
+#include <QMediaPlayer>
+
+#include "qgis_gui.h"
+#include "qgis_sip.h"
+
+class QLabel;
+class QVBoxLayout;
+class QVideoWidget;
+class QPushButton;
+class QSlider;
+
+/**
+ * \ingroup gui
+ * \brief The QgsMediaWidget class creates a widget for playing back audio and video media files.
+ */
+class GUI_EXPORT QgsMediaWidget : public QWidget
+{
+
+  public:
+
+    enum Mode
+    {
+      Audio,
+      Video,
+    };
+    Q_ENUM( Mode )
+
+    //! Constructor
+    explicit QgsMediaWidget( QWidget *parent SIP_TRANSFERTHIS = nullptr );
+
+    /**
+     * Returns the media path.
+     */
+    QString mediaPath() const { return mMediaPath; }
+
+    /**
+     * Sets the media \a path.
+     */
+    void setMediaPath( const QString &path );
+
+    /**
+     * Returns the media widget mode.
+     */
+    Mode mode() const { return mMode; }
+
+    /**
+     * Sets the media widget \a mode.
+     */
+    void setMode( Mode mode );
+
+    /**
+     * Returns the video frame's minimum height.
+     */
+    int videoMinimumHeight() const;
+
+    /**
+     * Sets the video frame's minimum height.
+     */
+    void setVideoMinimumHeight( int height );
+
+  private slots:
+
+    void mediaStatusChanged( QMediaPlayer::MediaStatus status );
+
+  private:
+
+    void adjustControls();
+    void toggleControls( bool enabled );
```

Comment:
Even if private, the name is misleading. Better to use something like `enableControls()` or `setControlsEnabled()`

Reply:
@uclaros , setControlsEnabled it is, thanks.

Reply:
@uclaros , done.